atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -651,13 +675,161 @@ def set_build_image(self):
+        elif image_from['kind'] == 'ImageStreamTag':
+            imagestream = image_from['name']
+        else:
+            raise UnknownKindException
+
+        return build_image, imagestream
+
+    def set_build_image(self, platforms):
+        """
+        Overrides build_image for worker, to be same as in orchestrator build
+        """
+        current_platform = platform.processor()
+        orchestrator_platform = current_platform or 'x86_64'
+
+        build_config_name = None
+        build_image = None
+        imagestream = None
+        current_buildimage = self.get_current_buildimage()
+        status = get_build_json().get("status", {})
+
+        # orchestrator platform is same as platform on which we want to built on,
+        # so we can use the same image
+        if platforms == set([orchestrator_platform]):
+            self.build_image_digests[orchestrator_platform] = current_buildimage
+            return
+
+        # BuildConfig exists
+        if 'config' in status:
+            config = status['config']
+            if config['kind'] == 'BuildConfig':
+                build_config_name = config['name']
+            else:
+                raise RuntimeError("Build config type isn't BuildConfig : %s" % config['kind'])
+
+        # get image build from build metadata, which is set for direct builds
+        # this is explicitly set by osbs-client, it isn't default OpenShift behaviour
+        else:
+            annotations = get_build_json().get("metadata", {}).get('annotations', {})
+            if 'from' in annotations:
+                scratch_from = json.loads(annotations['from'])
+
+                try:
+                    build_image, imagestream = self.process_image_from(scratch_from)
+                except UnknownKindException:
+                    raise RuntimeError("Build annotation has unknown 'kind' %s" %
+                                       scratch_from['kind'])
+            else:
+                raise RuntimeError("Build wasn't created from BuildConfig and neither"
+                                   " has 'from' annotation, which is needed for specified arch")
+
+        osbs = get_openshift_session(self.workflow, self.openshift_fallback)
+
+        # inspect BuildConfig
+        if build_config_name:
+            try:
+                build_config = osbs.os.get_build_config(build_config_name)
+            except OsbsException:
+                raise RuntimeError("Build config not found : %s" % build_config_name)
+
+            try:
+                build_from = build_config['spec']['strategy']['customStrategy']['from']
+            except KeyError:
+                raise RuntimeError("BuildConfig object is malformed")
+
+            try:
+                build_image, imagestream = self.process_image_from(build_from)
+            except UnknownKindException:
+                raise RuntimeError("BuildConfig object has unknown 'kind' %s" % build_from['kind'])
```

Comment:
Since you only get a build_config_name if config is in status, this block could be moved up to be part of the if config in status block, and then moved to the function I proposed earlier.

Reply:
mmm yeah

Reply:
I think you can tighten that up more:
build_image, imagestream = get_image_info_from_config()
has the current get_build_config_name and all of the code in the current if build_config_name: block

```
-       if build_config_name:
-       else:
+      if not (build_image or imagetream):
```